src/snapshot_middleware/lua.rs

```diff
@@ -20,36 +36,54 @@ pub fn snapshot_lua(
 ) -> anyhow::Result<Option<InstanceSnapshot>> {
     let file_name = path.file_name().unwrap().to_string_lossy();
 
-    let (class_name, instance_name) = if let Some(name) = match_trailing(&file_name, ".server.lua")
+    let is_run_context_enabled = true; //project.script_type == ScriptType::RunContext;
+
+    let run_context = &rbx_reflection_database::get().enums.get("RunContext").expect("Unable to get RunContext enums!").items;
+
+    let (script_type, instance_name) = if let Some(name) = match_trailing(&file_name, ".server.lua")
     {
-        ("Script", name)
+        (ScriptType::Server, name)
     } else if let Some(name) = match_trailing(&file_name, ".client.lua") {
-        ("LocalScript", name)
+        (ScriptType::Client, name)
     } else if let Some(name) = match_trailing(&file_name, ".lua") {
-        ("ModuleScript", name)
+        (ScriptType::Module, name)
     } else if let Some(name) = match_trailing(&file_name, ".server.luau") {
-        ("Script", name)
+        (ScriptType::Server, name)
     } else if let Some(name) = match_trailing(&file_name, ".client.luau") {
-        ("LocalScript", name)
+        (ScriptType::Client, name)
     } else if let Some(name) = match_trailing(&file_name, ".luau") {
-        ("ModuleScript", name)
+        (ScriptType::Module, name)
     } else {
         return Ok(None);
     };
 
+    let (class_name, run_context) = match (is_run_context_enabled, script_type) {
+        (true, ScriptType::Server) => ("Script", run_context.get("Server")),
+        (true, ScriptType::Client) => ("Script", run_context.get("Client")),
+        (false, ScriptType::Server) => ("Script", run_context.get("Legacy")),
+        (false, ScriptType::Client) => ("LocalScript", None),
+        (_, ScriptType::Module) => ("ModuleScript", None),
+    };
+
     let contents = vfs.read(path)?;
     let contents_str = str::from_utf8(&contents)
         .with_context(|| format!("File was not valid UTF-8: {}", path.display()))?
         .to_owned();
 
+    let mut properties = hashmap! {
+        "Source".to_owned() => contents_str.into(),
+    };
+
+    if let Some(run_context) = run_context {
+        properties.insert("RunContext".to_owned(), Enum::from_u32(run_context.to_owned()).into());
+    }
```

Comment:
This will force a reallocation every time when RunContext is enabled. We should probably swap away from the `hashmap!` macro as a result.

Reply:
The thing to keep in mind though, is that the RunContext could be "Server"/1, "Client"/2, or "Legacy"/0 so it has to be reassigned?